File: client/donations/src/app/donation/models/donation.ts
import{coin} from './coin'

export class Donation {
  public id:number;
  public isDeleting?:boolean

    constructor(
        public amount:number,
        public entityName: string,
        public entityType: string,
        public destiny: string,
        public coin: string,
        public exchangedRate: number,
        public conditions: string,
      ) {  }
}

File: server/donations/donations.DM/Models/Donation.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Text;

namespace donations.DM.Models
{
    public class Donation
    {
        [Key]
        public int Id { get; set; }
        public float Amount { get; set; }
        public string EntityName { get; set; }
        public Entity EntityType { get; set; }
        public string Destiny { get; set; }
        public string Conditions { get; set; }
        public Coin Coin { get; set; }
        public float ExchangeRate { get; set; }
    }
}
